specs/MultiProfile.md

```diff
@@ -328,6 +365,13 @@ interface ICoreWebView2Profile2 : ICoreWebView2Profile {
   /// See ICoreWebView2CookieManager.
   [propget] HRESULT CookieManager([out, retval] ICoreWebView2CookieManager** cookieManager);
 }
+
+[uuid(1c1ae2cc-d5c2-ffe3-d3e7-7857035d23b7), object, pointer_default(unique)]
+interface ICoreWebView2Profile3 : IUnknown {
+  /// Erase the profile completely. All webviews on this profile will be closed and  
+  /// the profile directory on disk will be deleted.
```

Comment:
That makes sense that its difficult to implement, but it seems like we have to have some way to tell the end developer that Delete is done and if it succeeded. Don't they need to know Delete has finished before closing their app so they don't close during the Delete? Or don't they need to know if Delete has finished before trying to create a new profile that might have the same name?

What is the scenario or use case for CoreWebView2Profile.Delete that the caller doesn't care if it succeeds or when it is done?

Reply:
The design has an update, and now is:

client call Delete API ->
send MOJOM message to the browser process -> 
browser close all WV2 and mark current profile as deleting -> 
client invoke render process  exited callback with reason WEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED 

1. The user call this Delete API, the API send a MOJOM message to the browser process.
2. The browser process calls API: profile_manager()->MaybeScheduleProfileForDeletion() to make the logic same as Edge: 
     - Mark current profile as "schedule for deleting" and will be truly deleted at thread exit. This makes the Delete API do not 
       need to wait for the local file to be deleted. So, the Delete Api could be treated as synchronous.
     - Close all WV2 under current profile.
3. Add a new Enum's value WEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED which indicate the render process 
    exited due to the related profile is deleted.
4. It will be failure when try to create a "schedule for deleting" profile, the behavior is same as Edge.

So, since the API is synchronous, it does not need a call-back for the caller.
The Delete API will not try to release the client's profile class and WV2 class, but them should not be used any more for the user after Delete API is called.

Reply:
11